Extensions/XMPPMUCSUB/XMPPMUCSUB.m

```diff
@@ -487,14 +490,18 @@ - (void)handleDiscoverFeaturesForRoomQueryIQ:(XMPPIQ *)iq withInfo:(XMPPBasicTra
                                                  code:[errorElem attributeIntegerValueForName:kXMPPCode
                                                                              withDefaultValue:0]
                                              userInfo:dict];
-            [self->multicastDelegate xmppMUCSUB:self failedToDiscoverFeaturesForRoomJID:roomJID withError:error];
+            if (roomJID) {
```

Comment:
If there is no `roomJID` we don't seem to do anything with the `errorElem` nor the `error` we've constructed. At the very least we should log as much info as we can perhaps? Also - is there any benefit in calling the `failedToDiscoverFeaturesForRoomJID` with a `nil` param for `roomJID`? And/or passing the `iq` and `info` as extra params?

Reply:
Maybe we should shove "no-jid" in there somehow?
```
-(XMPPJID *)validOrNullRoomJID:(XMPPJID *)roomJID
{
    return (roomJID ?: [XMPPJID jidWithString:@"NULL-ROOM-JID"]);
}
```

This could be used throughout...

Reply:
I changed the code to send error message to all delegates